### PR TITLE
lib: os: cbprintf: fix formatting bug

### PR DESCRIPTION
**Pull request by jeffkellam**
When logging with both alternative-forms and zero padding, place the zero padding after the alternative form prefix. 
As an example: `LOG_ERR("%#010x", 0xabcd);`
Should write `0x0000abcd` to the log and not `00000xabcd`

### CODE REVIEW

**Comment by carlescufi** [comment]
@jeffkellam thanks for the PR, please take a look at the commit formatting guidelines here:

https://docs.zephyrproject.org/latest/contribute/guidelines.html#commit-guidelines

**Comment by nordic-krch** [comment]
Please change commit prefix to `lib: os: cbprintf: ` since fix is in that library. Also, would be good to extend `tests/unit/cbprintf` test suite to cover that case.

**Comment by jeffkellam** [comment]
I've added the changes suggested by nordic-krch, and hopefully adjusted to be in compliance with the submission guidelines.

**Comment by andyross** [comment]
Code looks correct to me.  But still needs fixups to the commit message: your first commit (the actual fix) needs a Signed-off-by line and has needless indentation (format should be "title line", "blank line", "text", "blank line", "signoff").  Just read through "git log" for literally thousands of examples of how this works.  Yes, the rules are pedantic and needlessly overprecise, but in a big project things get really out of hand without rules.

Also the second patch should specify that it's adding test cases for the bug, not just restate the bug.

File: tests/unit/cbprintf/main.c [hunk 1]
```diff
@@ -1474,3 +1474,4 @@ void test_main(void)
 			 );
 	ztest_run_test_suite(test_prf);
 }
+
```
Comment (nordic-krch) [inline tests/unit/cbprintf/main.c]: not needed

File: tests/unit/cbprintf/main.c [hunk 2]
```diff
@@ -1141,6 +1141,56 @@ static void test_p(void)
 	zassert_equal(rc, 12, NULL);
 	zassert_equal(strncmp("/0x00cafe21/", buf, rc), 0, NULL);
 }
+static void test_alternative_form_hex(void)
```
Comment (nordic-krch) [inline tests/unit/cbprintf/main.c:1144]: commit message is wrong. Should be something like:
`tests: unit: cbprintf: Add test for ...`

Also the email address...

**Review by nordic-krch — COMMENTED** [review]
Commit message for change in `cbprintf_complete.c` should be updated:
- start with `lib: os: cbprintf: Fix ...`
- Do not mention logging here. Cbprintf is generic and can be used elsewhere. Just mention that following format specifier gives unexpected result (or something like that).

**Pull request closed without merging**